### PR TITLE
Update packages and fix tests

### PR DESCRIPTION
**Pull request by swissspidy**
The updated WP JS packages have been released today, including some of the fixes I've been working on regarding the build tooling.

Now anyone should be able to run the tests without having to run `npm link` or similar.

To verify:

1. Remove `node_modules`
1. Run `npm install`
1. Run `npm run test:js`

Expected output:

```
// ...

Test Suites: 1 skipped, 25 passed, 25 of 26 total
Tests:       6 skipped, 84 passed, 90 total
Snapshots:   6 passed, 6 total
Time:        22.123s
Ran all test suites.
```

Now that this is fixed we can look into running these tests on CI too.

Closes #2361.

### CODE REVIEW

**Comment by swissspidy** [comment]
Travis seems to fail because of a caching issue or something 🤔

**Comment by westonruter** [comment]
I tried clearing all caches and updating from `amp-stories-redux` but the build is still failing.

**Comment by westonruter** [comment]
Maybe it still depends on Chromium because failures are happening with `v8-compile-cache`.

**Comment by swissspidy** [comment]
Weird, now I am also getting the error locally. Looking.

**Comment by swissspidy** [comment]
At first glance it looks like the `@wordpress/dependency-extraction-webpack-plugin` package is incomplete/broken.

**Comment by swissspidy** [comment]
All tests are green 😻

File: includes/admin/class-amp-editor-blocks.php [hunk 1]
```diff
@@ -136,10 +136,15 @@ public function enqueue_block_editor_assets() {
 			);
 		}
 
+		$script_deps_path    = AMP__DIR__ . '/assets/js/amp-editor-blocks.deps.json';
```
Comment (westonruter) [inline includes/admin/class-amp-editor-blocks.php]: Problem: these JSON files aren't getting created when doing `npm run build`

File: tests/unit/setup-globals.js [hunk 2]
```diff
@@ -24,3 +24,26 @@ global.wp = {
...
+global.ampStoriesFonts = [
+	{
+		name: 'Arial',
+		slug: 'arial',
+	},
+	{
+		name: 'Roboto',
+		slug: 'roboto',
+		handle: 'roboto-font',
+		src: 'https://fonts.googleapis.com/css?family=Roboto%3A400%2C700&subset=latin%2Clatin-ext',
```
Comment (westonruter) [inline tests/unit/setup-globals.js]: Aside: we should consider using `display` at some point for fonts used in stories.

File: includes/admin/class-amp-editor-blocks.php (hunk 1 above)
Comment (swissspidy) [reply includes/admin/class-amp-editor-blocks.php]: Fixed in 9a9beca3089f20bf32eb72cadce65135d5edf2e8 and cc28abdf79b091aac3f7539f0f332d795a21363d

File: tests/unit/setup-globals.js (hunk 2 above)
Comment (swissspidy) [reply tests/unit/setup-globals.js]: Created #2367 to track this.

File: includes/admin/class-amp-editor-blocks.php (hunk 1 above)
Comment (westonruter) [reply includes/admin/class-amp-editor-blocks.php]: Good. The list file changes is now:

```
A	wp-content/plugins/amp/assets/js/amp-block-editor.deps.json
A	wp-content/plugins/amp/assets/js/amp-block-validation.deps.json
A	wp-content/plugins/amp/assets/js/amp-editor-blocks.deps.json
A	wp-content/plugins/amp/assets/js/amp-post-meta-box.deps.json
A	wp-content/plugins/amp/assets/js/amp-stories.deps.json
A	wp-content/plugins/amp/assets/js/amp-validated-url-post-edit-screen.deps.json
A	wp-content/plugins/amp/assets/js/amp-validated-urls-index.deps.json
A	wp-content/plugins/amp/assets/js/amp-validation-detail-toggle.deps.json
A	wp-content/plugins/amp/assets/js/amp-validation-single-error-url-details.deps.json
A	wp-content/plugins/amp/assets/js/amp-validation-tooltips.deps.json
A	wp-content/plugins/amp/assets/js/wp-dom-ready.deps.json
A	wp-content/plugins/amp/assets/js/wp-i18n.deps.json
D	wp-content/plugins/amp/assets/js/amp-admin-pointer.js
M	wp-content/plugins/amp/amp.php
M	wp-content/plugins/amp/assets/css/amp-stories-compiled-rtl.css
M	wp-content/plugins/amp/assets/css/amp-stories-compiled.css
M	wp-content/plugins/amp/assets/js/amp-block-editor.js
M	wp-content/plugins/amp/assets/js/amp-block-validation.js
M	wp-content/plugins/amp/assets/js/amp-editor-blocks.js
M	wp-content/plugins/amp/assets/js/amp-post-meta-box.js
M	wp-content/plugins/amp/assets/js/amp-stories.js
M	wp-content/plugins/amp/assets/js/amp-validation-tooltips.js
M	wp-content/plugins/amp/assets/js/wp-dom-ready.js
M	wp-content/plugins/amp/assets/js/wp-i18n.js
M	wp-content/plugins/amp/includes/admin/class-amp-editor-blocks.php
M	wp-content/plugins/amp/includes/admin/class-amp-post-meta-box.php
M	wp-content/plugins/amp/includes/class-amp-story-post-type.php
M	wp-content/plugins/amp/includes/sanitizers/class-amp-style-sanitizer.php
M	wp-content/plugins/amp/includes/validation/class-amp-validated-url-post-type.php
M	wp-content/plugins/amp/includes/validation/class-amp-validation-error-taxonomy.php
M	wp-content/plugins/amp/includes/validation/class-amp-validation-manager.php
```

**Pull request merged**